Redesign login

David-Development opened the pull request:
Redesign login process

- Use Activity instead of DialogFragment (make it full-screen)
- Use Nextcloud colors
- Focus on Single Sign On while providing fallback to manual login

Would love to get some feedback on this! 
@tobiasKaminsky @Unpublished @AndyScherzinger 

<img src="https://user-images.githubusercontent.com/4489723/55768491-7d26f000-5a53-11e9-8b53-4f30b2c81f1d.png" width="250px" />
<img src="https://user-images.githubusercontent.com/4489723/55768495-7f894a00-5a53-11e9-86a6-64442807b7bd.png" width="250px" />

AndyScherzinger commented:
Looking good @David-Development, just some very minor papercuts 👍 

* The button text should also be rather blue than black (we fixed that for the files app couple of days ago)
* The Input field coloring seems to be light blue for username and password while it is grey for the URL instead of light blue also
* in general (probabaly rather a new PR if decided to be done at all): The newest materail design guide allows for non-capitalized button text which is also what we now went for with the files app since this helps with readability (while it seems strange the first time you see it).

tobiasKaminsky commented:
In regard of the buttons, I would do the same like we have in our "first run":
![2019-04-09-090141](https://user-images.githubusercontent.com/5836855/55779615-31516680-5aa6-11e9-87ef-9a5e34896f26.png)

On your screenshot your primary button looks a bit grey

Unpublished commented on News-Android-App/src/main/res/layout/activity_login_dialog.xml:
```diff
@@ -0,0 +1,156 @@
+                android:layout_alignEnd="@+id/password_container"
+                android:contentDescription="@string/content_desc_show_password"/>
+
+            <androidx.appcompat.widget.AppCompatCheckBox
```
```suggestion
            <CheckBox
```

See https://developer.android.com/reference/android/support/v7/widget/AppCompatCheckBox
> This will automatically be used when you use CheckBox in your layouts and the top-level activity / dialog is provided by appcompat. You should only need to manually use this class when writing custom views.

Unpublished commented on News-Android-App/src/main/res/layout/activity_login_dialog.xml:
```diff
@@ -0,0 +1,156 @@
+            android:layout_marginStart="4dp"
+            android:background="@drawable/rounded_button"
+            android:textColor="@android:color/black"
+            android:text="Login using Nextcloud Files App" />
```
Should be extracted to a string resource.

Unpublished commented on News-Android-App/src/main/res/layout/activity_login_dialog.xml:
```diff
@@ -0,0 +1,156 @@
+            android:gravity="center"
+            android:textColor="@color/material_grey_100"
+            android:text="Manual login"/>
+
```
Should be extracted to a string resource.

David-Development commented:
@Unpublished It has been like this before. That is correct. Unfortunately changing this requires a bit more work as the API is trying to read the login information from the shared preferences. Therefore we have to store the login information in the preferences already before we know if the entered configuration is correct. As we want to drop the "normal" login anyways sometime in the future I wouldn't want to put too much effort into fixing this. 

@AndyScherzinger @tobiasKaminsky Thanks for your input! I added most of your suggestions (also switched to the new MaterialComponents theme..). Only problem is that I can't get the hint color working when it's not focused. Not sure what's wrong there.. As we want to get rid of this option at some point, I think it wouldn't be too bad if we just leave it like that. 

<img src="https://user-images.githubusercontent.com/4489723/55829549-44cff200-5ae5-11e9-9955-ef028c2d0b3c.png" width="300px"/>
<img src="https://user-images.githubusercontent.com/4489723/55829552-4699b580-5ae5-11e9-8ca7-c80c2b834554.png" width="300px"/>

AndyScherzinger commented:
Yeah, I guess that could be postponed 👍
Is there a reason why the first button has blue and bold text while the second button has black and normal weighted text?

Other than that it looks very nice!

David-Development commented:
@AndyScherzinger yes! because it's outdated and shouldn't be used anymore.. That's why I left the second button black..

Awesome! Thanks for the feedback! :)

tobiasKaminsky commented:
> Only problem is that I can't get the hint color working when it's not focused.

https://github.com/nextcloud/android/blob/ef45b6f11d9ee1a9bc695e98b0b20b25c7e7b43c/src/main/java/com/owncloud/android/utils/ThemeUtils.java#L403-L421

Maybe this helps?

AndyScherzinger commented:
@David-Development Awesome work! ❤️
